src/HDF5.jl

```diff
@@ -222,6 +222,16 @@ const H5T_NATIVE_DOUBLE   = read_const(:H5T_NATIVE_DOUBLE_g)
 # Library versions
 const H5F_LIBVER_EARLIEST = 0
 const H5F_LIBVER_LATEST   = 1
+# Constructed types (occurs at runtime)
+function make_float16()
+    FLOAT16 = h5t_copy(H5T_NATIVE_FLOAT)
+    h5t_set_fields(FLOAT16, 15, 10, 5, 0, 10)
+    h5t_set_size(FLOAT16, 2)
+    h5t_set_ebias(FLOAT16, 15)
+    h5t_lock(FLOAT16)
+    return FLOAT16
+end
+# const H5T_FLOAT16 = make_float16()  (in `__init__()`)
```

Comment:
To trigger this possible bug, it looks like you'd need to first load a library which creates a custom id dynamically (through a call to `H5Iregister()` I guess?), *before* `H5open()` is called.  I don't know whether the libhdf5 API contract prevents this from happening.

After that, load HDF5.jl into the same process and the ids may be different from when HDF5.jl was loaded in build.jl.